src/views/Player.vue

```diff
@@ -224,15 +224,18 @@ export default defineComponent({
       );
     }
 
-    function selectSeason(season: Season): void {
+    async function selectSeason(season: Season): Promise<void> {
       playerStore.SET_SELECTED_SEASON(season);
-      playerStore.loadGameModeStats({});
-      playerStore.loadRaceStats();
-      playerStore.loadMatches(1);
-      playerStore.loadPlayerStatsRaceVersusRaceOnMap(battleTag.value);
-      playerStore.loadPlayerStatsHeroVersusRaceOnMap(battleTag.value);
-      playerStore.loadPlayerMmrRpTimeline();
-      playerStore.loadPlayerGameLengths();
+      await Promise.all([
```

Comment:
Love the parallelization!
Did you check that there's no interdependence in between those calls?

Reply:
As far as I could tell there wasn't any. Bunch of Ctrl+R reloading and visually comparing prod vs local and it looked correct. The only one was the MMR graph depending on `loadGameModeStats` and `loadRaceStats`